File: training-data/nim/sieve_of_eratosthenes.nim
proc sieve(limit: int): seq[int] =
  var isComposite = newSeq[bool](limit + 1)
  var i = 2
  while i * i <= limit:
    if not isComposite[i]:
      var j = i * i
      while j <= limit:
        isComposite[j] = true
        j += i
    inc i
  for n in 2 .. limit:
    if not isComposite[n]:
      result.add(n)

echo sieve(50)

File: training-data/julia/sieve_of_eratosthenes.jl
function sieve(limit::Int)
    is_composite = falses(limit + 1)
    for i in 2:isqrt(limit)
        if !is_composite[i]
            for j in i*i:i:limit
                is_composite[j] = true
            end
        end
    end
    return [n for n in 2:limit if !is_composite[n]]
end

println(sieve(50))
